test/unit/yandex_browser_test.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class YandexTest < Minitest::Test
+  test "detects Yandex on iOS device" do
+    browser = Browser.new(Browser["YANDEX_BROWSER_IOS"])
+    assert browser.yandex?
+    assert browser.yandex_browser?
+    refute browser.safari?
+    refute browser.chrome?
```

Comment:
I am not strongly opinionated here on whether this should be true/false, however I lean more towards the side of false. Why? 

I would say the following statements are true: 

Yandex is a web browser
Yandex is my browser shown by https://www.whatismybrowser.com (when using Yandex, that is)
Yandex is not the same browser as Chrome
Yandex is based on Chromium

If I read the methods, I would assume that: 

`chrome?` is equivalent to saying `is_chrome` (standard in some other programming languages) 
`chromium_based?` is equivalent to saying `is_chromium_based` (is based on OSS Chromium) 
`webkit_based?` is equivalent to saying `is_webkit_based` (is based on the WebKit browser engine)

Reply:
I think it's more than that. What if I want to display a Chrome extension installer (either directly or via Google Play store) for Yandex users? If you drop the `chrome?` predicate then you have to do this for all browsers that are heavily based on Chrome. I'd bring back this behavior and if you need to do something specific for Yandex, go `yandex?` instead.

Does that make sense?

Reply:
Now, thinking a little bit about this, I don't know if I'm following this rule for every Chrome-based browser we're detecting. That said, it's probably fine `chrome?` to return `false`. At some point, make we should have something like `chrome_based?`.